### PR TITLE
[IMP] mail: notification sort messaging menu

### PR DESCRIPTION
**Pull request by phenix-factory**

### CODE REVIEW

File: addons/mail/static/tests/new/chat/chat_window_new_message_tests.js [hunk 1]
```diff
@@ -129,7 +129,7 @@ QUnit.test(
 
         // open channel-2
         await click(".o_menu_systray i[aria-label='Messages']");
-        await click(".o-mail-notification-item:nth-child(2)");
+        await click(".o-mail-notification-item .o-mail-notification-item-name:contains(channel-2)");
```
Comment (alexkuhn) [inline addons/mail/static/tests/new/chat/chat_window_new_message_tests.js:132]: 👍

File: addons/mail/static/tests/new/messaging_menu/notification_tests.js [hunk 2]
```diff
@@ -253,3 +253,34 @@ QUnit.test("non-failure notifications are ignored", async function (assert) {
...
+                model: "mail.channel",
+                res_id: mailChannelId1,
+            },
+            {
+                date: "2020-01-01 00:00:00",
+                model: "mail.channel",
+                res_id: mailChannelId2,
+            },
+        ]);
+        const { click } = await start();
```
Comment (alexkuhn) [inline addons/mail/static/tests/new/messaging_menu/notification_tests.js]: No need local destruct `click`

File: addons/mail/static/src/new/messaging_menu/messaging_menu.js [hunk 3]
```diff
@@ -72,10 +72,10 @@ export class MessagingMenu extends Component {
 
         const tab = this.state.tab;
         if (tab === "all") {
-            return previews;
+            return previews.reverse();
         }
         const target = this.tabToThreadType(tab);
-        return previews.filter((preview) => target.includes(preview.type));
+        return previews.filter((preview) => target.includes(preview.type)).reverse();
```
Comment (alexkuhn) [inline addons/mail/static/src/new/messaging_menu/messaging_menu.js]: Order was wrong in test so you just reversed the order, but this is still wrong: ordering of threads should be based on date of last message, not the order of insertion.

**Review by alexkuhn — CHANGES_REQUESTED** [review]
I have cleaned the test. I'll let do the sort that is properly based on date of last message.

File: addons/mail/static/src/new/messaging_menu/messaging_menu.js (hunk 3 above)
Comment (phenix-factory) [reply addons/mail/static/src/new/messaging_menu/messaging_menu.js]: Done ! 
Btw message.dateTime is already a dateTime object :)

**Comment by alexkuhn** [comment]
I took the opportunity to rename `dateTime` to `datetime` (and simplify naming of special `datetime` format)

**Pull request merged**